### PR TITLE
ign -> gz Header Migration : ign-msgs

### PR DESCRIPTION
**Pull request by methylDragon**
See https://github.com/ignition-tooling/release-tools/issues/711

Notes:
- `proto` message definitions for `gz` and `ignition` will be installed alongside each other, but ONLY THE `gz` proto messages will result in code generation for sources and headers.
- The entirety of the `ignition` headers are instantiated as redirection headers pointing to the `gz` version
- The `gz` headers use the `ignition` namespace, pending namespace migration

### CODE REVIEW

**Comment by methylDragon** [comment]
All green!

File: conf/CMakeLists.txt [hunk 1]
```diff
@@ -20,4 +20,4 @@ configure_file(
 
 # Install the yaml configuration files in an unversioned location.
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${IGN_DESIGNATION}${PROJECT_VERSION_MAJOR}.yaml
-  DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/ignition/)
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/gz/)
```
Comment (chapulina) [inline conf/CMakeLists.txt:23]: This breaks the `ign msg` CLI. I believe we haven't been migrating the install paths for these yaml files yet and `gz-tools` is still looking at `install/share/ignition`

**Review by chapulina — CHANGES_REQUESTED** [review]
It looks like the `import` calls inside downstream messages are broken with this PR. I tried to compile `gz-sim`'s `main` branch on top of this PR and it fails with:

```
/home/chapulina/dev_focal/ws_garden/build/ignition-gazebo7/src/msgs/peer_control.pb.cc:23:8: error: _PROTOBUF_INTERNAL_EXPORT_protobuf_ignit
ion_2fmsgs_2fheader_2eproto_ does not name a type; did you mean _PROTOBUF_INTERNAL_EXPORT_protobuf_gz_2fmsgs_2fheader_2eproto_?             
   23 | extern PROTOBUF_INTERNAL_EXPORT_protobuf_ignition_2fmsgs_2fheader_2eproto ::google::protobuf::internal::SCCInfo<2> scc_info_Header;       |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                           
      |        PROTOBUF_INTERNAL_EXPORT_protobuf_gz_2fmsgs_2fheader_2eproto
/home/chapulina/dev_focal/ws_garden/build/ignition-gazebo7/src/msgs/peer_control.pb.cc:50:51: error: _scc_info_Header_ is not a member of _p
rotobuf_ignition_2fmsgs_2fheader_2eproto_                                                                                                   
   50 |       &protobuf_ignition_2fmsgs_2fheader_2eproto::scc_info_Header.base,}};                                                                |                                                   ^~~~~~~~~~~~~~~                                       
/home/chapulina/dev_focal/ws_garden/build/ignition-gazebo7/src/msgs/peer_control.pb.cc: In function _void protobuf_peer_5fcontrol_2eproto::A
ddDescriptorsImpl()_:                                                                                                                       
/home/chapulina/dev_focal/ws_garden/build/ignition-gazebo7/src/msgs/peer_control.pb.cc:105:48: error: _AddDescriptors_ is not a member of _p
rotobuf_ignition_2fmsgs_2fheader_2eproto_                                                                                                   
  105 |   ::protobuf_ignition_2fmsgs_2fheader_2eproto::AddDescriptors();                                                                    
      |                                                ^~~~~~~~~~~~~~ 
```

I believe that's coming from [here](https://github.com/ignitionrobotics/ign-gazebo/blob/c756b607dfb6f5033d72074563137ad5fd5368c8/src/msgs/peer_control.proto#L22).

File: conf/CMakeLists.txt (hunk 1 above)
Comment (methylDragon) [reply conf/CMakeLists.txt]: https://github.com/ignitionrobotics/ign-msgs/pull/249/commits/69bdd6a8ba7c68cff6783ec93233102aaa49fc1e

File: include/ignition/msgs/detail/PointCloudPackedUtils.hh [hunk 2]
```diff
@@ -13,428 +13,6 @@
  * See the License for the specific language governing permissions and
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply include/ignition/msgs/detail/PointCloudPackedUtils.hh]: https://github.com/ignitionrobotics/ign-msgs/pull/249/commits/3ca459bec31e1f64a67f2f23ba9080e38669da41

File: include/gz/msgs/SuppressWarning.hh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * limitations under the License.
+ *
+ */
+
+#ifndef GZ_MSGS_SUPPRESSWARNING_HH_
+#define GZ_MSGS_SUPPRESSWARNING_HH_
+
+#include <gz/utils/SuppressWarning.hh>
+
+#pragma message("ign-msgs SuppressWarning is deprecated, use ign-utils")
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply include/gz/msgs/SuppressWarning.hh]: https://github.com/ignitionrobotics/ign-msgs/pull/249/commits/3632be288be3523511737a32d94a304d74aeefe6

[the remark this answers is not in the document]
Comment (methylDragon) [reply include/gz/msgs/SuppressWarning.hh]: And explicitly removed with https://github.com/ignitionrobotics/ign-msgs/pull/249/commits/1f09015badabc95d570cca49185db9ed109ff970

**Comment by methylDragon** [comment]
Going with the deferred hard-tock option (B) with `gz` headers directing to `ignition` for now, and `ignition` being the "canonical" message definitions at the moment.

image

Gazebo builds.

Once we're ready, and namespaces have been migrated, we will then move forward with hard-tocking every message (and message header) to `gz`.

**Review by chapulina — APPROVED** [review]
LGTM, this shouldn't break any downstream libraries as it is.

We'll come back for the hard tock later. I'd like to at least make the C++ redirection headers work, because that's what most people are using.

The proto namespaces are only used by people creating custom messages or doing some other kind of advanced usage of the messages. Ideally we'd tick-tick the change, but it's proving difficult, so I think those advanced users will have to deal with a hard-tock. Fortunately, the migration only requires a quick search-and-replace on their end.

File: include/ignition/msgs/Factory.hh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2016 Open Source Robotics Foundation
+ * Copyright (C) 2017 Open Source Robotics Foundation
```
[the remark this answers is not in the document]
Comment (methylDragon) [reply include/ignition/msgs/Factory.hh]: Fixed! https://github.com/gazebosim/gz-msgs/pull/249/commits/684b3166b581c8a9024f94b3b134bcfe7d3a1b3a

I'll merge once CI is green :raised_hands:

**Pull request merged**